src/error.rs

```diff
@@ -262,6 +269,85 @@ fn from_sslerror_ref(err: &ssl::Error) -> Error {
     }
 }
 
+#[cfg(feature = "snappy")]
+fn from_snap_error_ref(err: &::snap::Error) -> Error {
+    match err {
+        &::snap::Error::TooBig { given, max } => {
```

Comment:
This is a lot of code just to wrap the underlying error. We may want to consider switching to [`error_chain`](https://github.com/brson/error-chain) at some point.